### PR TITLE
Fix #1758: Add support for shm:// on Windows

### LINKED ISSUE #1758: Support Windows for `shm://` RzIO plugin
Seems instead of `shm_open()` API the [CreateFileMappingW function (memoryapi.h)](https://docs.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-createfilemappingw?redirectedfrom=MSDN) could be used.

See https://stackoverflow.com/questions/26114518/ipc-between-python-and-win32-on-windows-os for the example

### PR DESCRIPTION
**Pull request by GustavoLCR**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

* Add support for `shm://` on Windows
* Fix wrong buffer size reporting with `shm://`

**Test plan**

Tests pass

**Closing issues**

Closes #1758

### CODE REVIEW

**Comment by XVilka** [comment]
Please resend the PR from the branch inside the repository, also starting with `dist-`, to trigger all CI jobs.

File: librz/io/p/io_shm.c [hunk 1]
```diff
@@ -10,17 +10,24 @@
 #define __UNIX__ 0
 #endif
 
+#if __UNIX__ || __WINDOWS__ && !defined(__QNX__) && !defined(__HAIKU__)
```
Comment (XVilka) [inline librz/io/p/io_shm.c:13]: Did you mean `!defined(__WINDOWS__)` here?

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/io/p/io_shm.c [hunk 2]
```diff
@@ -35,7 +42,11 @@ static int shm__write(RzIO *io, RzIODesc *fd, const ut8 *buf, int count) {
 		(void)memcpy(shm->buf + io->off, buf, count);
 		return count;
 	}
+#if !defined(__WINDOWS__)
 	return write(shm->fd, buf, count);
+#else
+	return 0;
+#endif
```
Comment (ret2libc) [inline librz/io/p/io_shm.c:49]: Isn't it possible to write/read on Windows?

Comment (XVilka) [reply librz/io/p/io_shm.c:49]: It uses the path of `if (shm->buf) ...` But maybe some comment could be added.

**Comment by GustavoLCR** [comment]
Moved to #2028

File: librz/io/p/io_shm.c (hunk 2 above)
Comment (GustavoLCR) [reply librz/io/p/io_shm.c:49]: @ret2libc you cant read/write to the mapped file handle, the code will always use the memory mapped address instead.

**Pull request closed without merging**